source/News.cpp

```diff
@@ -98,14 +111,14 @@ bool News::IsEmpty() const
 
 
 
-// Check if this news item is available on the given planet.
-bool News::Matches(const Planet *planet) const
+// Check if this news item is available given the player's planet and conditions.
+bool News::Matches(const Planet *planet, const PlayerInfo &player) const
 {
 	// If no location filter is specified, it should never match. This can be
 	// used to create news items that are never shown until an event "activates"
 	// them by specifying their location.
 	// Similarly, by updating a news item with "remove location", it can be deactivated.
-	return location.IsEmpty() ? false : location.Matches(planet);
+	return location.IsEmpty() ? false : (location.Matches(planet) && toShow.Test(player.Conditions()));
```

Comment:
Should we just pass the `player` all the way from SpaceportPanel to here, then use `player.GetPlanet()` instead of passing the player reference and the planet pointer at the same time?